pkg/services/sqlstore/dashboard_provisioning.go

```diff
@@ -88,3 +91,28 @@ func UnprovisionDashboard(cmd *models.UnprovisionDashboardCommand) error {
 	}
 	return nil
 }
+
+func DeleteOrphanedProvisionedDashboards(cmd *models.DeleteOrphanedProvisionedDashboardsCommand) error {
+	return inTransaction(func(sess *DBSession) error {
+		var result []*models.DashboardProvisioning
+
+		convertedReaderNames := make([]interface{}, len(cmd.ReaderNames))
+		for index, readerName := range cmd.ReaderNames {
+			convertedReaderNames[index] = readerName
+		}
+
+		err := sess.NotIn("name", convertedReaderNames...).Find(&result)
+		if err != nil {
+			return err
+		}
+
+		for _, deleteDashCommand := range result {
+			err := deleteDashboard(&models.DeleteDashboardCommand{Id: deleteDashCommand.DashboardId}, sess)
```

Comment:
Would be great if we could move this deletion loop outside the `inTransaction` wrapper. This could create very large transactions without any clear upsides. I suggest we query for potential dashboards to delete without a transaction and then delete each dashboard in its own transition.

Reply:
Yes, it makes sense. I have changed the `DeleteOrphanedProvisionedDashboards` function behavior according to your suggestions. To make CI works, I also rebased the branch onto master.